sonaranalyzer-dotnet/src/SonarAnalyzer.CSharp/ControlFlowGraph/CSharpControlFlowGraphBuilder.cs

```diff
@@ -612,7 +642,12 @@ private Block BuildGotoStatement(GotoStatementSyntax statement, Block currentBlo
 
         #region Build switch
 
-        private Block BuildSwitchStatement(SwitchStatementSyntax switchStatement, Block currentBlock)
+        private Block BuildSwitchStatement(SwitchStatementSyntax switchStatement, Block currentBlock) =>
+            switchStatement.Sections.All(s => s.Labels.All(l => l.IsAnyKind(csharp6SwitchLabelKinds)))
+                ? BuildSwitchStatement_Legacy(switchStatement, currentBlock)
+                : BuildSwitchStatement_CSharp7(switchStatement, currentBlock);

```

Comment:
The pattern-enabled switch statements now takes into account the order of the case section declarations (the non-pattern switch doesn't), hence two completely different ways to handle.